### ISSUE TITLE
Update Megalinter to v6

### ISSUE THREAD

**Issue by wesley-dean-gsa** [GSA/usagov-benefits-eligibility#561]
# User Story

As a BEARS **engineer**, in order to **ensure we get the latest and most effective automated tests**, I would like **to bump MegaLinter from v5 to v6** .

Acceptance Criteria:

- [ ] the GitHub `megalinter` action has been updated to use the new version
- [ ] review current configuration and verify that it matches our use-cases
  - [ ] consider reviewing which linters are run and whether or not we can use a slimmed-down flavor

Definition of Done:

- [ ] Code complete
- [ ] Tests coverage is greater than team benchmark (90% goal)
- [ ] Security scans passed
- [ ] Acceptance Criteria is met and it works as expected
- [ ] Accessibility tested
- [ ] Build process and deployment is automated and repeatable
- [ ] Load testing/performance testing
- [ ] Self Documentation whenever possible
- [ ] Feature toggles if appropriate
- [ ] Deployed to staging
- [ ] Usability testing
- [ ] PR approved / Peer reviewed
- [ ] PO approved

**Comment by wesley-dean-gsa** [comment]
add concurrency protection:

```yaml
concurrency:
  group: ${{ github.ref }}-${{ github.workflow }}
  cancel-in-progress: true
```

**Comment by wesley-dean-gsa** [comment]
closes #562
